server/utils/commonFunc.go

```diff
@@ -0,0 +1,68 @@
+package utils
+
+import (
+	"errors"
+	"fmt"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
+func IsPassValid(password string) error {
+
+	if len(password) < 8 {
+		return errors.New("password is too short")
+
+	}
+	hasUpperCase := false
+	hasLowerCase := false
+	hasNumbers := false
+	hasSpecial := false
+
+	for _, char := range password {
+		if char >= 'A' && char <= 'Z' {
+			hasUpperCase = true
+		} else if char >= 'a' && char <= 'z' {
+			hasLowerCase = true
+		} else if char >= '0' && char <= '9' {
+			hasNumbers = true
+		} else if char >= '!' && char <= '/' {
+			hasSpecial = true
+		} else if char >= ':' && char <= '@' {
+			hasSpecial = true
+		}
```

Comment:
Can we use a regex here?

Reply:
yes sir we can use  regexPattern := `^(?=.*[0-9])(?=.*[^a-zA-Z0-9]).{8,}$`